aspnetcore/tutorials/signalr-blazor-webassembly.md

```diff
@@ -0,0 +1,268 @@
+---
+title: Use ASP.NET Core SignalR with Blazor WebAssembly
+author: guardrex
+description: Create a chat app that uses ASP.NET Core SignalR with Blazor WebAssembly.
+monikerRange: '>= aspnetcore-3.1'
+ms.author: riande
+ms.custom: mvc
+ms.date: 01/31/2020
+no-loc: [Blazor, SignalR]
+uid: tutorials/signalr-blazor-webassembly
+---
+# Use ASP.NET Core SignalR with Blazor WebAssembly
+
+By [Daniel Roth](https://github.com/danroth27) and [Luke Latham](https://github.com/guardrex)
+
+[!INCLUDE[](~/includes/blazorwasm-preview-notice.md)]
+
+This tutorial teaches the basics of building a real-time app using SignalR with Blazor WebAssembly. You learn how to:
+
+> [!div class="checklist"]
+> * Create a Blazor WebAssembly Hosted app project
+> * Add the SignalR client library
+> * Add a SignalR hub
+> * Add SignalR services and an endpoint for the SignalR hub
+> * Add Razor component code for chat
+
+At the end of this tutorial, you'll have a working chat app.
+
+[View or download sample code](https://github.com/aspnet/AspNetCore.Docs/tree/master/aspnetcore/tutorials/signalr-blazor-webassembly/samples/) ([how to download](xref:index#how-to-download-a-sample))
+
+## Prerequisites
+
+# [Visual Studio](#tab/visual-studio)
+
+[!INCLUDE[](~/includes/net-core-prereqs-vs-3.1.md)]
+
+# [Visual Studio Code](#tab/visual-studio-code)
+
+[!INCLUDE[](~/includes/net-core-prereqs-vsc-3.1.md)]
+
+# [Visual Studio for Mac](#tab/visual-studio-mac)
+
+[!INCLUDE[](~/includes/net-core-prereqs-mac-3.1.md)]
+
+# [.NET Core CLI](#tab/netcore-cli/)
+
+[!INCLUDE[](~/includes/3.1-SDK.md)]
+
+---
+
+## Create a Blazor WebAssembly Hosted app project
```

Comment:
```suggestion
## Create a Blazor WebAssembly hosted app project
```

Reply:
Let's make "hosted" the most general adjective.

```suggestion
## Create a hosted Blazor WebAssembly app project
```